### PR TITLE
add hash algorithm to envelope protobuf and ran make proto-vendor

### PR DESCRIPTION
**Pull request by naiming-zededa**
Signed-off-by: Naiming Shen <naiming@zededa.com>
added the enum of hash algorithm to the envelope protobuf
ran 'make proto-vendor' to commit all the changes

### CODE REVIEW

File: api/proto/auth/auth.proto [hunk 1]
```diff
@@ -11,11 +11,17 @@ message AuthBody {
...
+
 message AuthContainer {
     AuthBody authPayload = 1;      // envolope body, a marshalled protobuf data or it can be null
     // if the length of senderCertHash received is not 16 bytes, then the protobuf
     // message either is not AuthContainer type, or is corrupted. Otherwise, the
     // reciever may not have the sender's signing certificate
-    string   senderCertHash = 2;   // 16 bytes in length, 1st 16 char of sender siging cert sha256 hash
-    bytes    signatureHash = 3;    // signature of the sha256 hash of the payload
+    hashAlgorithm algo = 2;        // hash algorithm used for both payload for signing and sender Cert
+    string   senderCertHash = 3;   // 16 bytes in length, 1st 16 char of sender siging cert sha256 hash
```
Comment (eriknordmark) [inline api/proto/auth/auth.proto]: bytes not string I assume.

File: api/proto/auth/auth.proto [hunk 2]
```diff
@@ -11,11 +11,17 @@ message AuthBody {
     bytes  payload = 1;
 }
 
+enum hashAlgorithm {
+    HASH_NONE = 0;
+    HASH_SHA256 = 1;
```
Comment (eriknordmark) [inline api/proto/auth/auth.proto]: If we ever want a different number than 16 bytes than this style of enum names isn't helpful.
I think it makes sense to encode the number of bytes of hash in the hashAlgorithm (and change the comment for senderCertHash to refer to the algo and not 16).

File: api/proto/auth/auth.proto (hunk 1 above)
Comment (naiming-zededa) [reply api/proto/auth/auth.proto]: ok, changed to bytes

File: api/proto/auth/auth.proto (hunk 2 above)
Comment (naiming-zededa) [reply api/proto/auth/auth.proto]: makes sense. BTW, roman asked to make this two commits, not combined into this mass commit. I'll move to a different PR #596. Will close this one.

**Comment by naiming-zededa** [comment]
move to PR #596

**Pull request closed without merging**